mailbox/backup/src/test/java/org/apache/james/mailbox/backup/MailBoxAnnotationsExtraFieldTest.java

```diff
@@ -0,0 +1,283 @@
+                .isEqualTo((MailboxMessageFixture.ANNOTATION_1.getKey().asString() + ":" + MailboxMessageFixture.ANNOTATION_1.getValue().orElse("") + "%" +
+                        MailboxMessageFixture.ANNOTATION_2.getKey().asString() + ":" + MailboxMessageFixture.ANNOTATION_2.getValue().orElse("")
+                        ).getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    @Nested
+    class GetCentralDirectoryData {
+
+        @Test
+        void getCentralDirectoryDataShouldThrowWhenNoValue() {
+            MailBoxAnnotationsExtraField testee = new MailBoxAnnotationsExtraField();
+
+            assertThatThrownBy(() -> testee.getCentralDirectoryData())
+                    .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void getCentralDirectoryDataShouldReturnByteArraysOfAnnotation1() {
+            MailBoxAnnotationsExtraField testee = new MailBoxAnnotationsExtraField(MailboxMessageFixture.WITH_ANNOTATION_1);
+            assertThat(testee.getCentralDirectoryData())
+                    .isEqualTo((MailboxMessageFixture.ANNOTATION_1.getKey().asString() + ":" + MailboxMessageFixture.ANNOTATION_1.getValue().orElse("")).getBytes(StandardCharsets.UTF_8));
+        }
+
+        @Test
+        void getCentralDirectoryDataShouldReturnByteArraysOfAnnotation1And2() {
+            MailBoxAnnotationsExtraField testee = new MailBoxAnnotationsExtraField(MailboxMessageFixture.WITH_ANNOTATION_1_AND_2);
+            assertThat(testee.getCentralDirectoryData())
+                    .isEqualTo((annotation1Serialized + "%" +
+                            MailboxMessageFixture.ANNOTATION_2.getKey().asString() + ":" + MailboxMessageFixture.ANNOTATION_2.getValue().orElse("")
+                    ).getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    @Nested
+    class ParseFromLocalFileData {
+
+        @Test
+        void parseFromLocalFileDataShouldParseByteData() {
+            String bufferContent = annotation1Serialized;
+            MailBoxAnnotationsExtraField testee = new MailBoxAnnotationsExtraField(MailboxMessageFixture.WITH_ANNOTATION_1);
+            testee.parseFromLocalFileData(bufferContent
+                .getBytes(StandardCharsets.UTF_8), 0, sizeAnnotation1);
+            assertThat(testee.getValue()).contains(bufferContent);
+        }
+
+        @Test
+        void parseFromLocalFileDataShouldParseByteDataWhenOffsetSet() {
+            String bufferContent = annotation1Serialized;
+            int offset = MailboxMessageFixture.ANNOTATION_1.getKey().asString().length() + 1;
+            MailBoxAnnotationsExtraField testee = new MailBoxAnnotationsExtraField(MailboxMessageFixture.WITH_ANNOTATION_1);
+            testee.parseFromLocalFileData(bufferContent
+                    .getBytes(StandardCharsets.UTF_8), offset, sizeAnnotation1 - offset);
+            assertThat(testee.getValue()).contains(MailboxMessageFixture.ANNOTATION_1.getValue().orElse(""));
```

Comment:
you should never have any logic in tests.
`or` should trigger a warning in your mind